DEVELOPER_GUIDE.md

```diff
@@ -170,6 +171,15 @@ TOTAL 1207 79 93%
 
 You can run a combination of these by installing [poetry-exec-plugin](https://github.com/keattang/poetry-exec-plugin) once and using the `poetry exec coverage` shortcut.
 
+### Protobuf Class Generation
+
+Some Transport Requests use [Protobuf](https://protobuf.dev/) messages. These offer benefits including better reverse compatibility and easier cross-platform class generation. When using a new or updated proto definition (`.proto` file extension) you will need to compile it for use with python.
```

Comment:
It would be nice to explain what these are a little better, with maybe a link to where the files come from in OpenSearch core?

Reply:
> It would be nice to explain what these are a little better, 

The link to protobuf.dev is pretty clear on the messages.

> with maybe a link to where the files come from in OpenSearch core?

Good idea, I'll add.